testing.md

```diff
@@ -0,0 +1,125 @@
+# Testing your Node.js app
+An important part of writing any Node.js application is ensuring that your code is covered with rebust tests. This ensures that your application is functional, high quality, and as safe guarded from bugs as possible. 
+
+# Types of Testing
+
+## Unit Testing
+Unit Testing is the process of testing the smallest testable part of your application and are intended to assert that each unit of the software performs as designed. 
+Unit tests are usually used to validate that a function returns values correctly, and that it handles errors gracefully. This is accomplished by passing both correct and incorrect parameters to the function and assessing the output.
+
+You can generally think of these tests as human readible assertions. For example, consider the following function:
+
+```javascript
+  function addOne(number) {
+    return number + 1;
+  }
+```
+
+The kinds of questions we could consider when writing our tests include:
+- _Does `addOne` correctly return a number incremented by one?_
+- _What happens when `addOne` doesn't receives a parameter?_
+- _What happens when `addOne` receives a parameter that isn't a number?_
+
+These questions not only direct you in how to test your function but also inform the design of the function itself, such as adding error handling for incorrect parameter types. 
+
+The below example demonstrates how we can write a unit test, written using [Jest](#jest):
+```javascript
+test('1 + 1 to equal 2', () => {
+  const result = addOne(1);
+  expect(result).toBe(2);
+});
+```
+
+If `addOne` is working as expected, the value of `result` should be `2` and the test should therefore pass.
+
+## Intergration Testing
+Integration Testing is the process of testing the interactions between multiple functions. They are intended to ensure that the combination of functions produces the correct results or behaviour.
+
+Intergration tests are writen as test plans that outline a scenario, usually emulating the scripts or modules they are used in. Consider the below script:
```

Comment:
Good stuff to see here on this topic @dakebl, thanks for contributing!
If I may jump in - we should clarify that integration testing is a broad term and it isn't strictly between functions but also between classes and generic code components for example testing a repository and its integration with a database (here one may choose to spin up a real database or use an in-memory mock database).

I think the integration test example might not be the best to use but would happy to hear what you think. Instead of individually calling the separated functions and then asserting on their integrations, I see integration testing as testing a class or function or some other component without mocking its dependencies. So for example, if sum() would internally call some functions then when you don't mock them as you do in unit tests, that's where you actually test the integration of sum() and everything that is using inside.

Just my 2 cents and happy to hear feedback.